File: 2490-circular-sentence/2490-circular-sentence.py
class Solution:
    def isCircularSentence(self, sentence: str) -> bool:
        
        sentence = sentence.split()
        for i in range(len(sentence)-1):
            
            if sentence[i][-1] != sentence[i+1][0]:
                
                return False
            
        return True and sentence and sentence[-1][-1] == sentence[0][0]
                

File: 2490-circular-sentence/2490-circular-sentence.js
/**
 * @param {string} sentence
 * @return {boolean}
 */
var isCircularSentence = function(sentence) {
    
    return sentence.split(" ").reduce( function(check, word,index,arr){
        
        return check && (index !== arr.length-1 ? arr.at(index).at(-1) === arr.at(index+1).at(0) : arr.at(index).at(-1) === arr.at(0).at(0)) ;
        
    },true);
    
};